src/darwin/Framework/CHIP/templates/MTRClusters-src.zapt

```diff
@@ -52,56 +53,71 @@ using chip::SessionHandle;
+                to invoke our completion handler. This is not great from a
+                type-safety perspective, of course. }}
+            ^(id _Nullable value, NSError * _Nullable error) {
+              completionHandler(error);
+                  [workItem endWork];
+            },
+          {{/if}}
+          ^(ExchangeManager & exchangeManager, const SessionHandle & session, Cancelable * success, Cancelable * failure) {
+            chip::Optional<uint16_t> timedInvokeTimeoutMs;
+            ListFreer listFreer;
+            {{asUpperCamelCase parent.name}}::Commands::{{asUpperCamelCase name}}::Type request;
+            if (timedInvokeTimeoutMsParam != nil) {
+              timedInvokeTimeoutMs.SetValue(timedInvokeTimeoutMsParam.unsignedShortValue);
             }
-           {{/unless}}
-          {{/last}}
-        {{/chip_cluster_command_arguments}}
-
-        auto successFn = Callback<{{>callbackName}}CallbackType>::FromCancelable(success);
-        auto failureFn = Callback<DefaultFailureCallbackType>::FromCancelable(failure);
-        chip::Controller::{{asUpperCamelCase parent.name}}Cluster cppCluster(exchangeManager, session, self->_endpoint);
-        return cppCluster.InvokeCommand(request, successFn->mContext, successFn->mCall, failureFn->mCall, timedInvokeTimeoutMs);
-    });
-
-    [self.device setExpectedValues:expectedValues expectedValueInterval:expectedValueIntervalMs];
+            {{#if mustUseTimedInvoke}}
+            if (!timedInvokeTimeoutMs.HasValue()) {
+              timedInvokeTimeoutMs.SetValue(10000);
+            }
+            {{/if}}
+            {{#chip_cluster_command_arguments}}
+              {{#first}}
+                {{#unless (commandHasRequiredField parent)}}
+                if (params != nil) {
+                {{/unless}}
+              {{/first}}
+              {{>encode_value target=(concat "request." (asLowerCamelCase label)) source=(concat "params." (asStructPropertyName label)) cluster=parent.parent.name errorCode="return CHIP_ERROR_INVALID_ARGUMENT;" depth=0}}
+              {{#last}}
+                {{#unless (commandHasRequiredField parent)}}
+                }
+               {{/unless}}
+              {{/last}}
+            {{/chip_cluster_command_arguments}}
+
+            auto successFn = Callback<{{>callbackName}}CallbackType>::FromCancelable(success);
+            auto failureFn = Callback<DefaultFailureCallbackType>::FromCancelable(failure);
+            chip::Controller::{{asUpperCamelCase parent.name}}Cluster cppCluster(exchangeManager, session, self->_endpoint);
+            return cppCluster.InvokeCommand(request, successFn->mContext, successFn->mCall, failureFn->mCall, timedInvokeTimeoutMs);
+      });
+    };
+    workItem.readyHandler = readyHandler;
+    [self.device.asyncCallbackWorkQueue enqueueWorkItem:workItem];
+
+    if ([expectedValueIntervalMs compare:@(0)] == NSOrderedAscending) {
```

Comment:
What happens here when `expectedValueIntervalMs` is nil?

Reply:
Mmm this part was not written as I did in MTRDevice.mm - thanks for catching this. I'll fix this to be the same.

Reply:
Logic was changed in a previous commit along with MTRDevice.mm side.